packages/wrangler/templates/pages-dev-util.ts

```diff
@@ -45,7 +45,10 @@ function transformRoutingRuleToRegExp(rule: string): RegExp {
 	}
 
 	// /foo* => /foo.* => ^/foo.*$
-	transformedRule = `^${transformedRule.replace("*", ".*")}$`;
+	// /*.* => /*\.* => /.*\..* => ^/.*\..*$
+	transformedRule = `^${transformedRule
+		.replaceAll(/\./g, "\\.")
+		.replaceAll(/\*/g, ".*")}$`;
```

Comment:
This seems like there might be other gremlins hiding here... What happens if a rule has a `+`? A `()`?

Reply:
Yeah I managed to find the internal logic (in C++ 😅) for how this works in production. It is a handcrafted glob implementation.
Given that in that implementation there is no regular expression involved, we should be escaping any regular expression special chars.

Reply:
That being said, we have had this logic in place for many months with not many complaints; so I don't think it is a high priority to make this an exact replica of the internal logic right now.

So I'd like to land this to unblock my own work for now, if that's OK.